server/datastore/mysql/apple_mdm.go

```diff
@@ -4116,3 +4170,40 @@ VALUES
 
 	return nil
 }
+
+// ListIOSAndIPadOSToRefetch returns the UUIDs of iPhones/iPads that should be refetched.
+// It will return iOS/iPadOS (enrolled) devices with the following criteria:
+//   - No refetch command has been sent to the device, or
+//   - More than refetchInterval time has passed since last refetch command was acknowledged.
+func (ds *Datastore) ListIOSAndIPadOSToRefetch(ctx context.Context, interval time.Duration) (uuids []string, err error) {
+	// Exclude iPhones/iPads that already have a "refetch" command queued without an answer.
+	// The subquery is to get the last REFETCH command sent per device and its answer (if any).
+	stmt := fmt.Sprintf(`
+SELECT h.uuid FROM hosts h
+JOIN host_mdm hmdm ON hmdm.host_id = h.id
+LEFT JOIN (
+	SELECT last_refetch_commands.id, last_refetch_commands.command_uuid, r.status FROM (
+		SELECT q.id, q.command_uuid FROM nano_enrollment_queue q JOIN (SELECT q.id, q.priority, MAX(q.created_at) as max_created_at
+		FROM nano_enrollment_queue AS q
+		WHERE
+	  	  q.active = 1 AND LEFT(q.command_uuid, %d) = '%s'
```

Comment:
Looks like this sub-query on `last_refectch_commands` selects every available row that begins with "REFETCH" and groups them, just to double-check, does the row get updated to set `active = 0` once processed? If not this will only get bigger and slower with time. I'm not sure if we ever do that nor if we ever delete from the queue.

Reply:
I've pushed the changes that remove the optimization.